src/Symfony/Component/HttpFoundation/Tests/StreamedJsonResponseTest.php

```diff
@@ -57,6 +57,48 @@ public function testResponseWithoutAnGenerator()
         $this->assertSame('{"_embedded":{"articles":["Article 1","Article 2","Article 3"]}}', $content);
     }
 
+    public function testResponseWithPlaceholder()
+    {
+        // the placeholder must not conflict with generator injection
+        $content = $this->createSendResponse(
+            [
+                '_embedded' => [
+                    'articles' => $this->generatorArray('Article'),
+                    'placeholder' => '__symfony_json__',
+                    'news' => $this->generatorSimple('News'),
+                ],
+                'placeholder' => '__symfony_json__',
+            ],
+        );
+
+        $this->assertSame('{"_embedded":{"articles":[{"title":"Article 1"},{"title":"Article 2"},{"title":"Article 3"}],"placeholder":"__symfony_json__","news":["News 1","News 2","News 3"]},"placeholder":"__symfony_json__"}', $content);
+    }
+
+    public function testResponseWithMixedKeyType()
+    {
+        $content = $this->createSendResponse(
+            [
+                '_embedded' => [
+                    'list' => (function (): \Generator {
+                        yield 0 => 'test';
+                        yield 'key' => 'value';
+                    })(),
+                    'map' => (function (): \Generator {
+                        yield 'key' => 'value';
+                        yield 0 => 'test';
+                    })(),
+                    'integer' => (function (): \Generator {
+                        yield 1 => 'one';
+                        yield 3 => '3';
+                    })(),
+                ],
+            ],
+            \JSON_NUMERIC_CHECK
```

Comment:
```suggestion
            ]
```

Seems not to change anything and can be removed? Else we should create a dedicated test case for it.

Reply:
It does change the value `'3'` that is encoded as an integer. This check is related to key encoding. Numeric keys are the purpose of this test case. I don't want to add a dedicated test case for that, the Symfony test suite is already very long to run.

Reply:
we should create an own test case for it then.

Reply:
I moved the assertion to `testEncodingOptions`. This test ensure the encoding options are correctly used.